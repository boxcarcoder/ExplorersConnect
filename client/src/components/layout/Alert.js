import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const Alert = ({ alertsState }) => {
  if (alertsState !== null && alertsState.length > 0) {
    return alertsState.map(alert => (
      <div key={alert.id} className={`alert alert-${alert.alertType}`}>
        {alert.msg}
      </div>
    ));
  }

  return null;
};

Alert.propTypes = {
  alertsState: PropTypes.array.isRequired
};

//mapping the redux state (array of alerts in reducer) to a usable prop
const mapStateToProps = state => ({
  alertsState: state.alert //propName: state.reducer
});

export default connect(mapStateToProps)(Alert);
