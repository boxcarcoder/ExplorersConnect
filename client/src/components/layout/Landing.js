import React from 'react';

const Landing = () => {
  return (
    <section className='landing'>
      <div className='dark-overlay'>
        <div className='landing-inner'>
          <h1 className='large'>Explorers Connect</h1>
          <p className='lead'>
            Create an explorer profile to share your gear, passions, and
            favorite adventures.
          </p>
          <div className='buttons'>
            <a href='register.html' className='btn btn-primary'>
              Register
            </a>
            <a href='login.html' className='btn btn'>
              Login
            </a>
          </div>
        </div>
      </div>
      {/* <a
        style={{background-color:'black';color:'white';text-decoration:'none';padding:'4px 6px';font-family:'-apple-system , BlinkMacSystemFont, \"San Francisco\", \"Helvetica Neue\", Helvetica, Ubuntu, Roboto, Noto, \"Segoe UI\", Arial, sans-serif ';font-size:'8px';font-weight:'bold';line-height:'1';display:'inline-block';border-radius:'3px'}}
        href='https://unsplash.com/@wanderingteddybear?utm_medium=referral&amp;utm_campaign=photographer-credit&amp;utm_content=creditBadge'
        target='_blank'
        rel='noopener noreferrer'
        title='High-resolution photos from Ted Bryan Yu on Unsplash'
        className='credit-box-abs'
      >
        <span style='display:inline-block;padding:2px 3px;font-size:11px'>
          <i className='fas fa-camera-retro'></i>
        </span>
        <span style='display:inline-block;padding:2px 3px'>Ted Bryan Yu</span>
      </a> */}
    </section>
  );
};

export default Landing;
