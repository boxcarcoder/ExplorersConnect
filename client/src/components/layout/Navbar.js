import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import { setAlert } from '../../actions/alert';

//redux
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { logout } from '../../actions/auth';

export const Navbar = ({
  logout,
  authState: { isAuthenticated },
  setAlert,
}) => {
  const handleLogout = (e) => {
    logout();
    setAlert('Logged out succesfully', 'success');
  };

  const authenticatedLinks = (
    <h5>
      <ul>
        <li>
          <Link to='/profiles'>Explorers</Link>
        </li>
        <li>
          <Link to='/posts'>Posts</Link>
        </li>
        <li>
          <a onClick={(e) => handleLogout(e)} href='#!'>
            <i className='fas fa-sign-out-alt' /> <span>Logout</span>
          </a>
        </li>
      </ul>
    </h5>
  );

  const guestLinks = (
    <h5>
      <ul>
        <li>
          <Link to='/profiles'>Explorers</Link>
        </li>
        <li>
          <Link to='/posts'>Posts</Link>
        </li>
      </ul>
    </h5>
  );

  return (
    <nav className='navbar bg-dark'>
      <h1>
        <Link to='/'>
          <i className='fas fa-hiking'></i> ExplorersConnect
        </Link>
      </h1>
      <Fragment>{isAuthenticated ? authenticatedLinks : guestLinks}</Fragment>
    </nav>
  );
};

Navbar.propTypes = {
  authState: PropTypes.object.isRequired,
  logout: PropTypes.func,
};

const mapStateToProps = (state) => ({
  authState: state.auth,
});

export default connect(mapStateToProps, { logout, setAlert })(Navbar);
