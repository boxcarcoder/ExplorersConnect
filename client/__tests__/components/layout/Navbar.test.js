import React from 'react';
import { shallow } from 'enzyme';

// Import the non-connected component
import { Navbar } from '../../../src/components/layout/Navbar';

// Globals for tests
let wrapper;
let props;
let mockLogout = jest.fn();
let mockSetAlert = jest.fn();

describe('<Navbar /> component.', () => {
  test('Successfully renders.', () => {
    props = {
      logout: mockLogout,
      authState: {
        isAuthenticated: true,
      },
      setAlert: mockSetAlert,
    };

    wrapper = shallow(<Navbar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Renders link to the ');
});
