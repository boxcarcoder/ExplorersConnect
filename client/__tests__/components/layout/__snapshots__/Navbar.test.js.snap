// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> component. Successfully renders. 1`] = `
<nav
  className="navbar bg-dark"
>
  <h1>
    <Link
      to="/"
    >
      <i
        className="fas fa-hiking"
      />
       ExplorersConnect
    </Link>
  </h1>
  <h5>
    <ul>
      <li>
        <Link
          to="/profiles"
        >
          Explorers
        </Link>
      </li>
      <li>
        <Link
          to="/posts"
        >
          Posts
        </Link>
      </li>
      <li>
        <a
          href="#!"
          onClick={[Function]}
        >
          <i
            className="fas fa-sign-out-alt"
          />
           
          <span>
            Logout
          </span>
        </a>
      </li>
    </ul>
  </h5>
</nav>
`;
